functional_tests/steps/datasets.py

```diff
@@ -11,29 +11,28 @@
 def look_up_and_select_dataset(context: Context) -> None:
     mock_dataset = {
         "id": "example1",
+        "dataset_id": "example1",
         "description": "Example dataset for functional testing",
         "title": "Looked Up Dataset",
-        "version": 1,
-        "links": {
-            "latest_version": {
-                "href": "/datasets/example1/editions/example-dataset-1/versions/1",
-                "id": 1,
-            },
-        },
+        "version": "1",
+        "edition": "example-dataset-1",
+        "latest_version": {"id": "1"},
     }
```

Comment:
We should update mock to adhere to the new structure, which doesn't have "id" etc.